core/src/main/java/sunstone/core/SunstoneConfig.java

```diff
@@ -0,0 +1,155 @@
+package sunstone.core;
+
+
+import io.smallrye.config.SmallRyeConfig;
+import io.smallrye.config.SmallRyeConfigBuilder;
+import org.eclipse.microprofile.config.spi.ConfigSource;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Objects;
+import java.util.Properties;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+/**
+ * The class wraps and configure SmallRyeConfig so that it can resolve bare expressions and read sunstone.properties
+ */
+public class SunstoneConfig {
+
+    static final SmallRyeConfig config = new SmallRyeConfigBuilder()
+            .addDefaultInterceptors()
+            .addDefaultSources()
+            .addDiscoveredSources()
+            .withSources(new SunstonePropertiesSource(), new SunstoneExpressionSource())
+            .build();
+
+    /**
+     * Get property defined by the key and resolve it to the String.
+     */
+    public static String getString(String key) {
+        return config.getValue(key, String.class);
+    }
+
+    public static SmallRyeConfig unwrap() {
+        return config;
+    }
+
+    /**
+     * Delegate to {@link SmallRyeConfig#getValue(String, Class)}
+     */
+    public static <V> V getValue(String key, Class<V> clazz) {
+        return config.getValue(key, clazz);
+    }
+
+    /**
+     * Get property defined by the key. If not present, return default value. Type of the default value is also used as
+     * a converter in {@link SmallRyeConfig#getValue(String, Class)}
+     */
+    @SuppressWarnings("unchecked")
+    public static <V> V getValue(String key, V defaultValue) {
+        Objects.requireNonNull(defaultValue);
+        Object value = config.isPropertyPresent(key) ? config.getValue(key, defaultValue.getClass()) : defaultValue;
+        if (!defaultValue.getClass().isAssignableFrom(value.getClass())) {
+            throw new ClassCastException(String.format("Can not cast %s to %s", value, defaultValue.getClass()));
+        }
+        return (V) defaultValue.getClass().cast(value);
+    }
+
+    /**
+     * Resolve expression. Resolve ${.*} into other config properties.
+     *
+     * {@code clazz} is used as converter in {@link SmallRyeConfig#getValue(String, Class)} which is ultimately used for
+     * resolving expression.
+     */
+    public static <V> V resolveExpression(String expression, Class<V> clazz) {
+        return config.getValue(SunstoneExpressionSource.registerExpression(expression), clazz);
+    }
+
+
+    /**
+     * Resolve expression to String. If expression is blank, it is returned as is.
+     * Resolve ${.*} into other config properties.
```

Comment:
Is it possible mention example in javadoc? I don't get it :/

Reply:
Or when I want to use this method over `getString()` ? 

I as a programmer need to know if property can contain expression? If I know it does not contain expression use  `getString()` and if property can contain expression then to use `resolveExpressionToString()`?

By default I would expect  that all properties can contain expression.

Reply:
If it is so. Could we align names of those variants somehow?

```
Without Expression Resolution          - With Expression Resolution
==============================================================================
getString()                            - getStringWithExpressionResolution()
getStringWithoutExpressionResolution() - getString()
getStringWithoutExpressionResolution() - getStringWithExpressionResolution()
```

So it is clear they differs only in fact if they resolve expressions or no. Which pair will be picked depends on fact what we expect to be main usage. Names can be improved.

Reply:
This is an additional functionality for expression resolution. For example you have `vm-${my.system.property}`.

Getting a property having a key for it - expressions are automatically resolved by SR Config. In my usecase, I don't have the key. I have a value.

Reply:
Ok we called with Ivan and I understand now javadoc better. 

Now I understand main use case for this are strings in testsuite. Not values from key=value - getString() handles that by default.

Can that fact be added to javadoc? That would help me.

Reply:
Added.